toolkit/tools/pkg/imagecustomizerlib/customizeboot.go

```diff
@@ -46,8 +51,162 @@ const (
 
 	// The index of the SELinux mode value.
 	selinuxConfigModeRegexSELinuxMode = 1
+
+	// The index of the search command indent portion
+	searchCommandRegexIndentIndex = 1
+
+	// The start/end indexes of the kernel file path
+	linuxPathRegexPathStart = 2
+	linuxPathRegexPathEnd   = 3
+
+	// The start/end indexes of the initrd image file
+	initPathRegexPathStart    = 2
+	initPathRegexPathIndexEnd = 3
 )
 
+func setSearchCommand(inputGrubCfgContent string, searchCommand string) (outputGrubCfgContent string, err error) {
+
+	// Check if a 'search...' command is present.
+	searchMatch := searchCommandRegex.FindStringIndex(inputGrubCfgContent)
+	if searchMatch != nil {
+		// Find the search command indent.
+		subMatch := searchCommandRegex.FindStringSubmatch(inputGrubCfgContent)
+		indent := ""
+		if len(subMatch) > searchCommandRegexIndentIndex {
+			indent = subMatch[searchCommandRegexIndentIndex]
+		}
+		// Replace the search command.
+		start := searchMatch[0]
+		end := searchMatch[1]
+		outputGrubCfgContent = inputGrubCfgContent[:start] + indent + searchCommand + " " + inputGrubCfgContent[end:]
+	} else {
+		// If no search command already exists, insert the search command right
```

Comment:
The search command can work inside or outside the `menuentry` block. When the search is not present in the original grub.cfg, we just need the search command to appear right before the linux command because there are no other dependencies. This way, it's kept closest to where it is used. Is there a benefit from moving it outside the `menuentry`?